src/Faker/Provider/de_DE/Company.php

```diff
@@ -7,9 +7,31 @@ class Company extends \Faker\Provider\Company
     protected static $formats = array(
         '{{lastName}} {{companySuffix}}',
         '{{lastName}} {{lastName}} {{companySuffix}}',
+        '{{lastName}} {{companyPartner}} {{companySuffix}}',
+        '{{lastName}} {{companyPartner}}',
+        '{{branchePrefix}}{{lastName}}',
+        '{{lastName}} {{brancheSuffix}}',
+        '{{branchePrefix}}{{lastName}} {{companySuffix}}',
+        '{{lastName}} {{brancheSuffix}} {{companySuffix}}',
         '{{lastName}}',
         '{{lastName}}',
     );
 
     protected static $companySuffix = array('e.G.', 'e.V.', 'GbR', 'GbR', 'OHG mbH', 'GmbH & Co. OHG', 'AG & Co. OHG', 'GmbH', 'GmbH', 'GmbH', 'GmbH', 'AG', 'AG', 'AG', 'AG', 'KG', 'KG', 'KG', 'GmbH & Co. KG', 'GmbH & Co. KG', 'AG & Co. KG', 'Stiftung & Co. KG', 'KGaA', 'GmbH & Co. KGaA', 'AG & Co. KGaA', 'Stiftung & Co. KGaA');
+
+    protected static $partner = array('und Partner', 'und Söhne', '& Partner', '& Söhne');
```

Comment:
I reckon a static method that returns a random element from this array is missing, take a look at the [Company provider](https://github.com/fzaninotto/Faker/blob/master/src/Faker/Provider/Company.php)